test/e2e/specs/editor/plugins/align-hook.spec.js

```diff
@@ -0,0 +1,326 @@
+		await requestUtils.activatePlugin( 'gutenberg-test-align-hook' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin( 'gutenberg-test-align-hook' );
+	} );
+
+	test.describe( 'Block with no alignment set', () => {
+		test( 'shows no alignment buttons on the alignment toolbar', async ( {
+			editor,
+			page,
+		} ) => {
+			await editor.insertBlock( { name: 'test/test-no-alignment-set' } );
+			await editor.showBlockToolbar();
+
+			await expect(
+				page
+					.getByRole( 'toolbar', { name: 'Block tools' } )
+					.getByRole( 'button', { name: 'Align', exact: true } )
+			).toBeHidden();
+		} );
+
+		test( 'does not save any alignment related attribute or class', async ( {
+			editor,
+		} ) => {
+			await editor.insertBlock( { name: 'test/test-no-alignment-set' } );
+
+			await expect
+				.poll( editor.getEditedPostContent )
+				.not.toContain( '"align":' );
+		} );
+	} );
+
+	test.describe( 'Block with align true', () => {
+		test( 'shows the expected buttons on the alignment toolbar', async ( {
+			editor,
+			page,
+		} ) => {
+			await editor.insertBlock( { name: 'test/test-align-true' } );
+			await editor.clickBlockToolbarButton( 'Align' );
+
+			await expect(
+				page
+					.getByRole( 'menu', { name: 'Align' } )
+					.getByRole( 'menuitemradio' )
+			).toHaveText( Object.values( alignLabels ) );
+		} );
+
+		test( 'applies none alignment by default', async ( {
+			editor,
+			page,
+		} ) => {
+			await editor.insertBlock( { name: 'test/test-align-true' } );
+			await editor.clickBlockToolbarButton( 'Align' );
+
+			await expect(
+				page
+					.getByRole( 'menu', { name: 'Align' } )
+					.getByRole( 'menuitemradio', { name: 'None' } )
+			).toHaveAttribute( 'aria-checked', 'true' );
+		} );
+
+		test( 'correctly applies the selected alignment and correctly removes the alignment', async ( {
+			editor,
+			page,
+		} ) => {
+			await editor.insertBlock( { name: 'test/test-align-true' } );
+			await editor.clickBlockToolbarButton( 'Align' );
+
+			const dropdown = page.getByRole( 'menu', {
+				name: 'Align',
+			} );
+
+			await dropdown
+				.getByRole( 'menuitemradio', { name: 'Align right' } )
+				.click();
+
+			// Verify the button of the specified alignment is pressed.
+			await editor.clickBlockToolbarButton( 'Align' );
+			await expect(
+				dropdown.getByRole( 'menuitemradio', { name: 'Align right' } )
+			).toHaveAttribute( 'aria-checked', 'true' );
+
+			// Verify the markup of the selected alignment was generated.
+			let markup = await editor.getEditedPostContent();
+			expect( markup ).toContain( 'alignright' );
+
+			let blocks = await page.evaluate(
+				( [ content ] ) => {
+					return window.wp.blocks.parse( content );
+				},
+				[ markup ]
+			);
+			expect( blocks ).toMatchObject( [
+				{
+					name: 'test/test-align-true',
+					isValid: true,
+				},
+			] );
```

Comment:
The other `correctly applies the selected alignment and correctly removes the alignment` should also have similar assertions, but I've only included it here for now.

I think testing intentions here are more apparent than in the original tests. But it still seems odd to test block serialization/parsing here. WDYT?

Original code: https://github.com/WordPress/gutenberg/blob/cc1870a49f57f89cb4c97b41782420a36f77e4da/packages/e2e-tests/specs/editor/plugins/align-hook.test.js#L122-L127.

Reply:
From a user perspective, what happens if a block is invalid? What does the user see?

Reply:
On the theme side, they will see a broken layout. In the editor, when markup is invalid and the block has no deprecation handlers in the editor, they will see a notice.

Reply:
Hmm, it would be difficult to wait until the notice is not shown, 😅 but maybe we could use some other user-facing stuff there instead of the raw markup?

Reply:
Maybe the assertion that re-parsing markup is still valid should be performed by low-level tests. The [`full-content.test.js`](https://github.com/WordPress/gutenberg/blob/trunk/test/integration/full-content/full-content.test.js) integration test already does something similar for core blocks.

Here, we can ensure that the generated markup contains (or does not)  the alignment classes. What do you think?

Cc-ing @ntsekouras and @jorgefilipecosta as they worked on the original tests.

Reply:
I think it makes perfect sense 👍 .

Reply:
I've updated the assertions in [add8857](https://github.com/WordPress/gutenberg/pull/56480/commits/add8857095f264f829f31285990a7f22d554c03b).